src/app/common/grade-icon/grade-icon.component.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { GradeIconComponent } from './grade-icon.component';
+import { gradeService } from 'src/app/ajs-upgraded-providers';
+
+describe('GradeIconComponent', () => {
+  let component: GradeIconComponent;
+  let fixture: ComponentFixture<GradeIconComponent>;
+  let gradeServiceStub: jasmine.SpyObj<any>;
+
+  beforeEach(
+    waitForAsync(() => {
+      gradeServiceStub = {
+        grades: [
+          'Pass',
+          'Credit',
+          'Distinction',
+          'High Distinction'
+        ],
+        gradeAcronyms: {
+          'Fail': 'F',
+          'Pass': 'P',
+          'Credit': 'C',
+          'Distinction': 'D',
+          'High Distinction': 'HD',
+          0: 'P',
+          1: 'C',
+          2: 'D',
+          3: 'HD',
+        }
+      };
+
+      gradeServiceStub.grades[-1] = 'Fail';
+      gradeServiceStub.gradeAcronyms[-1] = "F";
```

Comment:
**thought (non-blocking):** This is a bit of an odd use of the array data structure. An array starts from `0` and this breaks the convention. I understand that this is ported straight from the `grade-service.coffee` but might be an area of improvement in a future PR.

Reply:
You're right. I assumed it's like this so that any invalid grade given to the service will return with Fail. Good idea to improve in the future.